scripts/synthetix/facilicate_trade.js

```diff
@@ -0,0 +1,88 @@
+  rinkeby: "0x0647b2C7a2a818276154b0fC79557F512B165bc1",
+}
+
+module.exports = async callback => {
+  try {
+    const snxjs = new SynthetixJs({ networkId: await web3.eth.net.getId() })
+    const exchange = await getExchange(web3)
+    const defaultAccount = (await web3.eth.getAccounts())[0]
+
+    // Both of these hardcoded tokens are assumed to have 18 decimal places.
+    // We "trust" that this will always be the case although it seems
+    // that synthetix reserves the authority to upgrade their token
+    // This could mean issuing a new one with a different number of decimals.
+    const sETHKey = await snxjs.sETH.currencyKey()
+    const sUSDKey = await snxjs.sUSD.currencyKey()
+
+    // Fetch token IDs and other relevant token information for order placement.
+    const referenceTokenId = (await exchange.tokenAddressToIdMap.call(sUSDAddress[process.env.NETWORK_NAME])).toNumber()
+    const etherTokenId = (await exchange.tokenAddressToIdMap.call(sETHAddress[process.env.NETWORK_NAME])).toNumber()
+
+    const tokenInfoPromises = fetchTokenInfoFromExchange(exchange, [referenceTokenId, etherTokenId])
+    const stableToken = await tokenInfoPromises[referenceTokenId]
+    const targetToken = await tokenInfoPromises[etherTokenId]
+
+    const buyTokens = [etherTokenId, referenceTokenId]
+    const sellTokens = [referenceTokenId, etherTokenId]
+
+    // Compute Rates and Fees based on price of sETH.
+    // Note that sUSD always has a price of 1 within synthetix protocol.
+    const exchangeRate = await snxjs.ExchangeRates.rateForCurrency(sETHKey)
+    const formatedRate = snxjs.utils.formatEther(exchangeRate)
+    console.log("sETH Price", snxjs.utils.formatEther(exchangeRate))
+    
+    // Using synthetix's fees, and formatting their return values with their tools, plus parseFloat.
+    const sETHTosUSDFee = parseFloat(
+      snxjs.utils.formatEther(
+        await snxjs.Exchanger.feeRateForExchange(sETHKey, sUSDKey)
+      )
+    )
+    const sUSDTosETHFee = parseFloat(
+      snxjs.utils.formatEther(
+        await snxjs.Exchanger.feeRateForExchange(sUSDKey, sETHKey)
+      )
+    )
+
+    // Compute buy-sell amounts based on unlimited orders with rates from above.
+    const [lowerBuyAmount, lowerSellAmount] = calculateBuyAndSellAmountsFromPrice(
+      formatedRate * (1 - sUSDTosETHFee),
+      stableToken,
+      targetToken
+    )
+
+    const [upperSellAmount, upperBuyAmount] = calculateBuyAndSellAmountsFromPrice(
+      formatedRate * (1 + sETHTosUSDFee),
+      stableToken,
+      targetToken
+    )
+    const buyAmounts = [lowerBuyAmount, upperBuyAmount]
+    const sellAmounts = [lowerSellAmount, upperSellAmount]
+
+    // Fetch auction index and declare validity interval for orders.
+    const batch_index = (await exchange.getCurrentBatchId.call()).toNumber()
+    const validFroms = Array(2).fill(batch_index)
+    const validTos = Array(2).fill(batch_index + 1)
+
+    // TODO - use replaceOrder if possible.
```

Comment:
Also @fleupold, as you rightfully pointed out, `replaceOrders` is nothing but cancellation followed by new order placement.

https://github.com/gnosis/dex-contracts/blob/b075120c985d4ad725ab8aaf6a325500907868ca/contracts/BatchExchange.sol#L273-L284

This will also be of interest to an issue in `dex-web`(?) regarding a feature request I once made about toping up an existing order that has already been filled. Ahh yes, it was about "Renewing Expired Orders" https://github.com/gnosis/dex-react/issues/389.

Of course, this is still of interest to a front end, but technically does not correspond to updating the `validTo` field of an existing order.